### DISCUSSION TITLE
[API Request] Support Multiple Album Versions

### DISCUSSION THREAD

**Discussion by jeffgt14** [opensubsonic/open-subsonic-api#79] [API requests]
### Type of change

API extension

### Proposal description

Expose a `VERSION` tag to add disambiguation amongst multiple releases of the same album.
i.e. [Dark Side of the Moon, Dark Side of the Moon (30th Anniversary Remaster), Dark Side of the Moon (40th Anniversary Remaster), Dark Side of the Moon (Quadrophonic Vinyl), Dark Side of the Moon (MFSL CD), etc.]

MusicBrainz ID's are great to keep the albums organized separately but the information exposing "what's different" is not there. Also, at least for the niche group of us into vinyl ripping, it's useful to see custom information such as WHO in the vinyl ripping internet community ripped the vinyl as well since it's not an exact science such as a CD rip.

Roon and Ampache currently use the custom `VERSION` tag for this information and I believe Navidrome may use this in some capacity as well.

### Backward compatibility

- [X] No backward compatibility impact.

### API details

I'm definitely not knowledgeable in this aspect :)

**Comment by epoupon** [comment]
This is exactly what the ReleaseGroup MBID is for. See for example https://musicbrainz.org/release-group/f5093c06-23e3-404f-aeaa-40f72885ee3a

**Reply by jeffgt14** [reply]
That's at a higher parent level. Picture below is stolen from a Navidrome thread but shows what I'm talking about. All of these albums are grouped under the same ReleaseGroup with the same Album name with different MBIDs. There is no additional field to disambiguate which one may be the Japan CD releases which likely has a bonus track and a US CD release. Catalog information isn't always consistent here either because sometimes the same catalog number may have various mastering.

![134478920-6ca0558f-0913-4c59-826f-b4ce3ef456b1](https://github.com/opensubsonic/open-subsonic-api/assets/34482770/e35b5b7e-2acd-4a66-90fd-ddf2f48ca57e)

**Reply by epoupon** [reply]
Even if we add support for the `version` custom tag, we still need a way to identify groups of releases.
For your example, another artist may have released an `Oceanborn` album, and we want an irrefutable way of excluding it (rather than, for example, looking to see if all the other tags are compatible with the release group).

**Reply by jeffgt14** [reply]
Maybe I'm not understanding how everything works. I am not suggesting we change how we identify groups of releases. Simply better identify the versions of albums in that release group. Roon does this very well by reading the `Version` tag. Below is an example of how the same album release is identified by reading this tag

https://community.roonlabs.com/t/album-versions-issue/152114/8

![03ecc3b915c6ffbdbd95e23f7080b126bf4e2c58](https://github.com/opensubsonic/open-subsonic-api/assets/34482770/ecae799c-c0c2-4150-9420-851b17e84297)

**Reply by epoupon** [reply]
My bad, I just wanted to say we first need the mb release group id, which is not exposed too.
By the way, I can't find any picard plugin that populates the version tag. And I don't find any plugin that uses the disambiguation info from MB. How do people tag this in an automated way?

**Reply by jeffgt14** [reply]
Ok that would make sense too. I figure this would work without the MBID but I'm not 100% sure how it all works.

Disambiguation info should be able to be pulled from MB with this script:
`$set(Version,%_releasecomment%)`

This is per Ampache documentation
https://github.com/ampache/ampache/wiki/ampache6-details

I use puddletag to set Version off the folder name so I haven't personally tested the tagger script.

**Reply by dweymouth** [reply]
I would think the version (or edition, or whatever we decide on) tag would be something that most people would populate manually, if they care about it. As @jeffgt14 currently does with manual version info appended to the album title (discussed in Navidrome discord)

**Reply by sentriz** [reply]
personally I automatically set it with beets. it writes to the "Musicbrianz Album Comment" tag which is a concatenation on the mb release and release group disambiguations

**Comment by sentriz** [comment]
as @epoupon it would be nice to group releases in a release group, but for now it would probably be enough to just expose a `disambiguation` or `version` field. i think that would be backwards compatible if we did release groups, since both releases and release groups can have disambiguations
